extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMap.kt

```diff
@@ -44,18 +50,24 @@ class MapboxCarMap(
    * The carContext can also be found in the [MapboxCarMapObserver] callbacks. Make sure to
    * call [MapboxCarMap.clearObservers] when your car session is destroyed.
    */
-  val carContext: CarContext = mapInitOptions.context as? CarContext
-    ?: throw IllegalArgumentException("You must construct a MapboxCarMap with the CarContext")
-
-  private val carMapSurfaceOwner = CarMapSurfaceOwner()
-  private val carMapSurfaceCallback = CarMapSurfaceCallback(
-    carContext,
-    carMapSurfaceOwner,
-    mapInitOptions
-  )
+  val carContext: CarContext by lazy {
+    carMapSurfaceOwner.carContext
+  }
 
-  init {
-    carMapSurfaceCallback.onBind()
+  /**
+   * Calls [AppManager.setSurfaceCallback] to give Mapbox access to render the map onto Android
+   * Auto head units. You should only call this once per [CarContext] in a [Session].
+   *
+   * @param carContext Android Auto CarContext
+   * @param mapInitOptions parameter to initialize the head unit map
+   */
+  fun setup(
+    carContext: CarContext,
+    mapInitOptions: MapInitOptions,
+  ) = apply {
+    mapInitOptions.verifyCarContext()
+    carMapSurfaceOwner.setup(carContext, mapInitOptions)
+    carContext.getCarService(AppManager::class.java).setSurfaceCallback(carMapSurfaceOwner)
```

Comment:
Moved this service call up a layer to make it more visible. This isn't a functional change, but a small opinion that this should be clear that you are setting the service callback.